src/CaptureFile/CaptureFileTest.cpp

```diff
@@ -112,6 +112,34 @@ class CaptureFileTest : public CaptureFileHeaderTest {
     capture_file_ = std::move(capture_file_or_error.value());
   }
 
+  void VerifySectionReadable(uint64_t section_index, size_t section_size) {
+    EXPECT_EQ(capture_file_->GetSectionList()[section_index].size, section_size);
+    std::vector<char> data;
+    data.reserve(section_size);
+    const ErrorMessageOr<void> read_result =
+        capture_file_->ReadFromSection(section_index, 0, data.data(), section_size);
+    EXPECT_THAT(read_result, HasNoError());
+  }
+
+  void VerifySingleUserDataSectionExistsAtEnd(size_t user_data_section_size) {
+    ASSERT_GE(capture_file_->GetSectionList().size(), 1);
+
+    const uint64_t last_section_index = capture_file_->GetSectionList().size() - 1;
+
+    const CaptureFileSection section_list_entry =
+        capture_file_->GetSectionList()[last_section_index];
+    EXPECT_EQ(section_list_entry.type, kSectionTypeUserData);
+    EXPECT_EQ(section_list_entry.size, user_data_section_size);
```

Comment:
This is a bit redundant as we verify that "the user data section is at the end" and "the size is correct" later in
```
    EXPECT_EQ(capture_file_->FindAllSectionsByType(kSectionTypeUserData).front(),
              last_section_index);

    VerifySectionReadable(last_section_index, user_data_section_size);
```